sdk/Javascript/src/MobileServiceSyncContext.js

```diff
@@ -0,0 +1,70 @@
+﻿// ----------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// ----------------------------------------------------------------------------
+
+/// <reference path="C:\Program Files (x86)\Microsoft SDKs\Windows\v8.0\ExtensionSDKs\Microsoft.WinJS.1.0\1.0\DesignTime\CommonConfiguration\Neutral\Microsoft.WinJS.1.0\js\base.js" />
+/// <reference path="C:\Program Files (x86)\Microsoft SDKs\Windows\v8.0\ExtensionSDKs\Microsoft.WinJS.1.0\1.0\DesignTime\CommonConfiguration\Neutral\Microsoft.WinJS.1.0\js\ui.js" />
+/// <reference path="Generated\MobileServices.DevIntellisense.js" />
+
+var Validate = require('Validate');
+var Platform = require('Platform');
+var MobileServiceMemoryStore = require('MobileServiceMemoryStore').MobileServiceMemoryStore;
+
+function MobileServiceSyncContext(client) {
+    /// <summary>
+    /// Creates an instance of the MobileServiceSyncContext class
+    /// </summary>
+    /// <param name="client" type="MobileServiceClient" mayBeNull="false">
+    /// The MobileServiceClient used to make requests.
+    /// </param>
+
+    Validate.notNull(client, 'client');
+
+    this.client = client;
+
+    this.initialize = function (store) {
+        /// <summary>
+        /// Initializes the sync context with an instance of the store to be used
+        /// </summary>
+
+        if (store === undefined) {
+            this.store = new MobileServiceMemoryStore();
```

Comment:
this.store is not defined anywhere, also since private should be _store, _client, etc